Tests/Indicators/TimeSeriesIndicatorTests.cs

```diff
@@ -0,0 +1,71 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using Accord.Math;
+using NUnit.Framework;
+using QuantConnect.Indicators;
+
+namespace QuantConnect.Tests.Indicators
+{
+    [TestFixture]
+    public class TimeSeriesIndicatorTests
+    {
+        [Test]
+        public void DifferencesAndUndifferencesSeries()
+        {
+            var test = new double[100];
+            for (var i = 0; i < test.Length; i++)
+            {
+                switch (i % 2)
+                {
+                    case 0:
+                        test[i] = 1;
+                        break;
+                    default:
+                        test[i] = 2;
+                        break;
+                }
+            }
+
+            double[] heads;
+            var differencer = TimeSeriesIndicator.DifferenceSeries(1, test, out heads);
+            Assert.AreEqual(-1, differencer.Sum());
+            Assert.AreEqual(test.Sum(), TimeSeriesIndicator.InverseDifferencedSeries(differencer, heads).Sum());
```

Comment:
Cool! `InverseDifferencedSeries` should have it's own unit test 👍, each unit test should assert a single component.
Think could move the following into a static private helper method in this test class:
```
var test = new double[100];
for (var i = 0; i < test.Length; i++)
{
    switch (i % 2)
    {
        case 0:
            test[i] = 0;
            break;
        default:
            test[i] = 1;
            break;
    }
}
```